### PR TITLE
Added missing functions to Workplane

### PR DESCRIPTION
**Pull request by adam-urbanczyk**

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
New sweep option does not work with the current implementation of sweep. Check how FreeCAD does this currently:

https://github.com/FreeCAD/FreeCAD/blob/master/src/Mod/Part/App/PartFeatures.cpp#L413

**Comment by dcowden** [comment]
wow, thats some pretty impenetrable code

**Comment by dcowden** [comment]
@adam-urbanczyk pipeshell and left in the FreeCAD based CQ was flakey too, but i don't remember seeing this particular one.  We commonly saw failures to generate, and weird behaviors at the boundaries.

That said, i use Solidworks and Onshape frequently, and even they have a hard time getting things like this right.  I usually have to provide a guide curve to get shapes like i want in those packages. Based on that experience, it might be unreasonable to assume that the default behavior should 'just work' without extra information, in the form of either guide paths, or boundary constraints

**Comment by adam-urbanczyk** [comment]
OK, I will stop investigating this for now then. BTW: OCC `BRepOffsetAPI_MakePipeShell` API provides a lot of options - it would be nice to have more understanding of those at some point in the future.

File: tests/TestCadQuery.py [hunk 1]
```diff
@@ -1326,6 +1452,80 @@ def testExplicitClean(self):
...
+            .rect(1.5, 3.5, forConstruction=True).vertices().cskHole(0.125, 0.25, 82, depth=None)
+        self.saveModel(result)
+
+        # bottom plane
+        s = Workplane(Plane.bottom())
+        result = s.rect(2.0, 4.0).extrude(0.5).faces(">Z").workplane()\
+            .rect(1.5, 3.5, forConstruction=True).vertices().cskHole(0.125, 0.25, 82, depth=None)
+        self.saveModel(result)
+
+    def testIsIndide(self):
```
Comment (jmwright) [inline tests/TestCadQuery.py]: I think this should be `testIsInside`

**Comment by jmwright** [comment]
@adam-urbanczyk Is this ready for review/approval, or is there more you want to add?

**Comment by adam-urbanczyk** [comment]
@jmwright ready for review

File: tests/TestCadQuery.py (hunk 1 above)
Comment (adam-urbanczyk) [reply tests/TestCadQuery.py]: Fixed in b59e277

File: cadquery/occ_impl/shapes.py [hunk 2]
```diff
@@ -1364,22 +1364,49 @@ def sweep(cls, outerWire, innerWires, path, makeSolid=True, isFrenet=False):
...
+    def sweep_multi(cls, profiles, path, makeSolid=True, isFrenet=False):
+        """
+        Multi section sweep. Only single outer profile per section is allowed.
+
+        :param profiles: list of profiles
+        :param path: The wire to sweep the face resulting from the wires over
+        :return: a Solid object
+        """
+        if path.ShapeType() == 'Edge':
+            path = Wire.assembleEdges([path, ])
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py]: Is there supposed to be another item in the list after `path`?

Comment (adam-urbanczyk) [reply cadquery/occ_impl/shapes.py]: No, just a convention I use (analogous to 1-tuples (el,) ).  Should I change this?

Comment (jmwright) [reply cadquery/occ_impl/shapes.py]: I think it's fine. I just wanted to confirm that it was intentional.

Comment (adam-urbanczyk) [reply cadquery/occ_impl/shapes.py]: OK

**Comment by adam-urbanczyk** [comment]
@dcowden any comments on the PR? If not I'd like to merge.

**Comment by dcowden** [comment]
Looks great to me! Though, at this point I think you guys know cq better than I do!

**Pull request merged**